crates/router/src/connector/adyen/transformers.rs

```diff
@@ -289,139 +352,341 @@ impl TryFrom<&types::BrowserInformation> for AdyenBrowserInfo {
-            currency: item.request.currency.to_string(),
-            value: item.request.amount,
-        };
-        let ccard = match item.request.payment_method_data {
-            api::PaymentMethod::Card(ref ccard) => Some(ccard),
-            api::PaymentMethod::BankTransfer
-            | api::PaymentMethod::Wallet(_)
-            | api::PaymentMethod::PayLater(_)
-            | api::PaymentMethod::Paypal => None,
-        };
+        match item.payment_method {
+            storage_models::enums::PaymentMethodType::Card => get_card_specific_payment_data(item),
+            storage_models::enums::PaymentMethodType::PayLater => {
+                get_paylater_specific_payment_data(item)
+            }
+            storage_models::enums::PaymentMethodType::Wallet => {
+                get_wallet_specific_payment_data(item)
+            }
+            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
+        }
+    }
+}
 
-        let wallet_data = match item.request.payment_method_data {
-            api::PaymentMethod::Wallet(ref wallet_data) => Some(wallet_data),
-            _ => None,
-        };
+fn get_shopper_interaction_data(
+    item: &types::PaymentsAuthorizeRouterData,
+) -> AdyenShopperInteraction {
+    match item.request.off_session {
+        Some(true) => AdyenShopperInteraction::ContinuedAuthentication,
+        _ => AdyenShopperInteraction::Ecommerce,
+    }
+}
 
-        let shopper_interaction = match item.request.off_session {
-            Some(true) => AdyenShopperInteraction::ContinuedAuthentication,
-            _ => AdyenShopperInteraction::Ecommerce,
-        };
+fn get_return_url(
+    item: &types::PaymentsAuthorizeRouterData,
+) -> Result<String, error_stack::Report<errors::ConnectorError>> {
+    item.router_return_url
+        .clone()
+        .ok_or_else(utils::missing_field_err("router_return_url"))
+}
 
-        let recurring_processing_model = match item.request.setup_future_usage {
-            Some(storage_enums::FutureUsage::OffSession) => {
-                Some(AdyenRecurringModel::UnscheduledCardOnFile)
-            }
-            _ => None,
-        };
+fn get_recurring_processing_model(
+    item: &types::PaymentsAuthorizeRouterData,
+) -> Option<AdyenRecurringModel> {
+    match item.request.setup_future_usage {
+        Some(storage_enums::FutureUsage::OffSession) => {
+            Some(AdyenRecurringModel::UnscheduledCardOnFile)
+        }
+        _ => None,
+    }
+}
 
-        let payment_type = match item.payment_method {
-            storage_enums::PaymentMethodType::Card => "scheme".to_string(),
-            storage_enums::PaymentMethodType::Wallet => wallet_data
-                .get_required_value("wallet_data")
-                .change_context(errors::ConnectorError::RequestEncodingFailed)?
-                .issuer_name
-                .to_string(),
-            _ => "None".to_string(),
-        };
+fn get_browser_info(
```

Comment:
@jagan-jaya the getter methods which we have implemented in utils validate the given field and throws error in case of  None. But here i don't need to validate for None as "AdyenBrowserInfo" is an optional field. So i am keeping it here itself.